db/system_keyspace.cc

```diff
@@ -2506,15 +2506,17 @@ future<service::topology> system_keyspace::load_topology_state() {
                 }
                 ret.req_param.emplace(host_id, service::rebuild_param{*rebuild_option});
                 break;
-            case service::topology_request::join:
-                ret.req_param.emplace(host_id, service::join_param{num_tokens});
-                break;
             default:
                 // no parameters for other requests
                 break;
             }
         } else {
             switch (nstate) {
+            case service::node_state::bootstrapping:
+                // We need to know how many tokens to generate (we only generate them
+                // at some point of the sequence)
```

Comment:
What do you mean by "we only generate them at some point of the sequence"?

Reply:
To bootstrap the node, the topology coordinator has to go through several steps, states of the topology coordinator (that's what I meant by "sequence" here). Here I meant that generating tokens doesn't happen at the first step like it does now but a bit later.

I'll try to rephrase to make it clearer.